### ISSUE TITLE
ML Preprocessing Demos

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/oceanbench#8]
**TLDR**

Showcase how the dataset can be used for difference scenarios.

**Case Studies**

* Interpolation: Batch of Coordinates
* Interpolation: Batch of patches, `x=ssh_sims(lat x lon x time), y=ssh_obs(lat x lon x time)`
* Surrogate: Batch of patches, `x=ssh(lat x lon x t-1), y=ssh(lat x lon x t)`
* Patch-ify, e.g. `[h,w] -> [P, h//factor, w//factor]`
* Cube-ify, e.g. `[h,w,c] -> [P, c//factor, h//factor, w//factor]`
* Coordinate-ify, e.g. `xr.dataset -> pd.DataFrame`, `[LAT,LON,TIME,SSH] -> [ssh,lat,lon,time] x (latxlonxtimex1)`
  * `broadcast_like`
  * `ravel`
* storage suited for local sampling!, e.g. 1) storage for frequency sampling, 2) storage for local sampling

---

- *Interpolation*: Implicit/Coordinate-Based methods, e.g. `GPs, NerFs`
    - `x=[lat,lon,time]`
    - `f=[ssh,sst]`
- *Interpolation*: Explicit/Coordinate-Based methods, e.g. `CNN`
    - `f=u(lat x lon x time x var)`
    - `x=(lat,lon,time,mask)`
- *Inverse Problems*: Operator Methods, e.g. `4DVar`
    - `f=u_sim(lat x lon x time x vars)`
    - `y=u_obs(lat x lon x time x vars)`
    - `x=(lat,lon,time,mask)`
- *Predictions Problems*: Functional Methods, e.g. `GPs, NerFs`
    - `f=[var1,var2,...]`
    - `c=[var1,var2,...]`
    - `x=[lat,lon,time]`
- *Predictions Problems*: Operator Methods, e.g. `FNO, MNO`
    - `f=u(lat x lon x time x var)`
    - `g=a(lat x lon x time x var)`
    - `x=(lat,lon,time,mask)`

---
**Needed Functionality**

- [ ] Load ML Ready Data
- [ ] Define slicing (Patch-ify)
- [ ] Redefine Grid
